src/core/system_parameters/Beidou_B1c.h

```diff
@@ -0,0 +1,156 @@
+/*!
+ * \file Beidou_B1c.h
+ * \brief Defines system parameters for Beidou B1c signal
+ * \author Andrew Kamble, 2019. andrewkamble88@gmail.com 
+ * \note Code added as part of GSoC 2019 program
+ * Detailed description of the file here if needed.
+ *
+ * -------------------------------------------------------------------------
+ *
+ * Copyright (C) 2010-2019  (see AUTHORS file for a list of contributors)
+ *
+ * GNSS-SDR is a software defined Global Navigation
+ *          Satellite Systems receiver
+ *
+ * This file is part of GNSS-SDR.
+ *
+ * GNSS-SDR is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * GNSS-SDR is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GNSS-SDR. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * -------------------------------------------------------------------------
+ */
+
+#ifndef GNSS_SDR_BEIDOU_B1c_H_
+#define GNSS_SDR_BEIDOU_B1c_H_
+
+
+
+const double BEIDOU_B1cd_CODE_RATE_HZ = 10.23e6;      //!< BEIDOU_B1c data code rate [chips/s]
+const int32_t BEIDOU_B1cd_CODE_LENGTH_CHIPS = 10230;  //!< BEIDOU_B1c data  code length [chips]
+
+
+const double BEIDOU_B1cp_CODE_RATE_HZ = 10.23e6;      //!< BEIDOU_B1c pilot code rate [chips/s]
+const int32_t BEIDOU_B1cp_CODE_LENGTH_CHIPS = 10230;  //!< BEIDOU_B1c pilot code length [chips]
+
+//!<Beidou B1c secondary codes. Data component do not heave secondary code. Pilot component have secondary codes which are generated by truncating Weil Codes. 
+const int32_t BEIDOU_B1cp_SECONDARY_CODE_LENGTH_CHIPS = 1800;  // Each secondary code of Pilot component has the length 1800 chips.
+
+
+const int32_t WEIL_N = 10243;
```

Comment:
Consider adding 'BEIDOU_B1C_' or similar in front of these variables in order to achieve consistency with variables within the file.